program_info/win_install.nsi

```diff
@@ -0,0 +1,223 @@
+!insertmacro MUI_LANGUAGE "Estonian"
+!insertmacro MUI_LANGUAGE "Turkish"
+!insertmacro MUI_LANGUAGE "Lithuanian"
+!insertmacro MUI_LANGUAGE "Slovenian"
+!insertmacro MUI_LANGUAGE "Serbian"
+!insertmacro MUI_LANGUAGE "SerbianLatin"
+!insertmacro MUI_LANGUAGE "Arabic"
+!insertmacro MUI_LANGUAGE "Farsi"
+!insertmacro MUI_LANGUAGE "Hebrew"
+!insertmacro MUI_LANGUAGE "Indonesian"
+!insertmacro MUI_LANGUAGE "Mongolian"
+!insertmacro MUI_LANGUAGE "Luxembourgish"
+!insertmacro MUI_LANGUAGE "Albanian"
+!insertmacro MUI_LANGUAGE "Breton"
+!insertmacro MUI_LANGUAGE "Belarusian"
+!insertmacro MUI_LANGUAGE "Icelandic"
+!insertmacro MUI_LANGUAGE "Malay"
+!insertmacro MUI_LANGUAGE "Bosnian"
+!insertmacro MUI_LANGUAGE "Kurdish"
+!insertmacro MUI_LANGUAGE "Irish"
+!insertmacro MUI_LANGUAGE "Uzbek"
+!insertmacro MUI_LANGUAGE "Galician"
+!insertmacro MUI_LANGUAGE "Afrikaans"
+!insertmacro MUI_LANGUAGE "Catalan"
+!insertmacro MUI_LANGUAGE "Esperanto"
+!insertmacro MUI_LANGUAGE "Asturian"
+!insertmacro MUI_LANGUAGE "Basque"
+!insertmacro MUI_LANGUAGE "Pashto"
+!insertmacro MUI_LANGUAGE "ScotsGaelic"
+!insertmacro MUI_LANGUAGE "Georgian"
+!insertmacro MUI_LANGUAGE "Vietnamese"
+!insertmacro MUI_LANGUAGE "Welsh"
+!insertmacro MUI_LANGUAGE "Armenian"
+!insertmacro MUI_LANGUAGE "Corsican"
+!insertmacro MUI_LANGUAGE "Tatar"
+!insertmacro MUI_LANGUAGE "Hindi"
+
+;--------------------------------
+
+; The stuff to install
+Section "PolyMC"
+
+  SectionIn RO
+
+  nsExec::Exec /TIMEOUT=2000 'TaskKill /IM polymc.exe /F'
+
+  SetOutPath $INSTDIR
+
+  File "polymc.exe"
+  File "qt.conf"
+  File *.dll
+  File /r "iconengines"
+  File /r "imageformats"
+  File /r "jars"
+  File /r "platforms"
+  File /r "styles"
+
+  ; Write the installation path into the registry
+  WriteRegStr HKCU Software\PolyMC "InstallDir" "$INSTDIR"
+
+  ; Write the uninstall keys for Windows
+  !define UNINST_KEY "Software\Microsoft\Windows\CurrentVersion\Uninstall\PolyMC"
+  WriteRegStr HKCU "${UNINST_KEY}" "DisplayName" "PolyMC"
+  WriteRegStr HKCU "${UNINST_KEY}" "DisplayIcon" "$INSTDIR\polymc.exe"
+  WriteRegStr HKCU "${UNINST_KEY}" "UninstallString" '"$INSTDIR\uninstall.exe"'
+  WriteRegStr HKCU "${UNINST_KEY}" "QuietUninstallString" '"$INSTDIR\uninstall.exe" /S'
+  WriteRegStr HKCU "${UNINST_KEY}" "InstallLocation" "$INSTDIR"
+  WriteRegStr HKCU "${UNINST_KEY}" "Publisher" "PolyMC Contributors"
+  WriteRegStr HKCU "${UNINST_KEY}" "ProductVersion" "${VERSION}"
+  ${GetSize} "$INSTDIR" "/S=0K" $0 $1 $2
+  IntFmt $0 "0x%08X" $0
+  WriteRegDWORD HKCU "${UNINST_KEY}" "EstimatedSize" "$0"
+  WriteRegDWORD HKCU "${UNINST_KEY}" "NoModify" 1
+  WriteRegDWORD HKCU "${UNINST_KEY}" "NoRepair" 1
+  WriteUninstaller "$INSTDIR\uninstall.exe"
+
+SectionEnd
+
+Section "Start Menu Shortcuts"
+
+  CreateShortcut "$SMPROGRAMS\PolyMC.lnk" "$INSTDIR\polymc.exe" "" "$INSTDIR\polymc.exe" 0
+
+SectionEnd
+
+Section /o "Portable"
+
+  SetOutPath $INSTDIR
+  File "portable.txt"
+
+SectionEnd
```

Comment:
I feel like the `Portable` checkbox might cause confusion, as its consequences aren't explained anywhere. It also doesn't make much sense in this case, as it probably doesn't make sense registering a portable application like any installed application.

Reply:
I did that originally so that portable installs would be able to use that installer to update themselves, since most Windows update libraries need an installer to run. But then I realized that even installing the “regular” version on top won’t affect the `portable.txt`, so that shouldn’t be a problem.

What might be a problem is that there may need to be a workaround for when portable PolyMC installations want to do an update by running the installer. It would register them as an installed app upon an update in the installer’s current state, even if they weren’t registered as such before.